module/core/src/Infrastructure/Validator/UnitNameUniqueValidator.php

```diff
@@ -44,19 +45,16 @@ public function validate($value, Constraint $constraint): void
             throw new UnexpectedTypeException($constraint, UnitNameUnique::class);
         }
 
-        if (null === $value || '' === $value) {
-            return;
+        if (!$value instanceof UnitFormModel) {
```

Comment:
In my opinion, you should create just one validator and call it UnitFormValidator as it's validating the model itself - now the logic is scattered around different validators though it does not bring additional value